core/src/main/scala/cats/eq/Is.scala

```diff
@@ -0,0 +1,64 @@
+package cats.eq
+
+import cats.Id
+
+/**
+ * A value of `A Is B` witnesses that the types `A` and `B` are the same.
+ * You can use this value to coerce `A` into `B` and therefore `A Is B` is
+ * similar in feel to `A =:= B`; however, `A Is B` has more computational
+ * content and offers greater functionality.
+ *
+ * `A Is B` is also known as Leibniz equality and represents computationally
+ * the idea that two things are equal if they take equal values on all
+ * predicates. In other words, `A` equals `B` when for all predicates `P[_]`
+ * we have that `P[A]` implies `P[B]` and visa versa. `A Is B` represents
+ * this idea computationally providing exploitable structure to our equalities.
+ *
+ * The heart of `A Is B` is the `subst` method which converts values `F[A]`
+ * into values `F[B]` for any `F[_]`. Creative choices of `F` enable type
+ * equalities to be substituted into many more complex type expressions.
+ */
+abstract class Is[A, B] {
```

Comment:
should this not be `sealed`? Is there any case where you should ever make a new subclass of this outside of this file?